js/piwik.js

```diff
@@ -6424,6 +6449,116 @@ if (typeof window.Matomo !== 'object') {
+              return true;
+            }
+
+          /**
+           * Calling this method will remember that the user has given consent for a given feature across multiple requests by setting
+           * a cookie. You can optionally define the lifetime of that cookie in hours using a parameter.
+           *
+           * When you call this method, we imply that the user has given consent for the given feature, and will also
+           * imply consent for all future page views unless the cookie expires (if timeout defined) or the user
+           * deletes all their cookies. This means even if you call {@link requireFeatureConsent(featureList)}, then the given feature will still work.
+           *
+           * Please note that this feature requires you to set the `cookieDomain` and `cookiePath` correctly and requires
+           * that you do not disable cookies. Please also note that when you call this method, consent will be implied
+           * for all sites that match the configured cookieDomain and cookiePath. Depending on your website structure,
+           * you may need to restrict or widen the scope of the cookie domain/path to ensure the consent is applied
+           * to the sites you want.
+           *
+           * @param {int} hoursToExpire After how many hours the consent should expire. By default the consent is valid
+           *                          for ~30 years unless cookies are deleted by the user or the browser prior to this
+           * @returns {boolean} Returns true if 'remember consent given for the feature' was stored successfully.
+           */
+            this.rememberConsentGivenForFeature = function (feature, hoursToExpire) {
+              var msToExpire;
+              if (hoursToExpire) {
+                msToExpire = hoursToExpire * 60 * 60 * 1000;
+              } else {
+                msToExpire = 30 * 365 * 24 * 60 * 60 * 1000;
+              }
+
+              if (!this.setConsentGivenForFeature(feature)) {
+                return false;
+              }
+
+              var now = new Date().getTime();
+              setCookie(getCookieName('consent_feature_'+feature), now, msToExpire, configCookiePath, configCookieDomain, configCookieIsSecure, configCookieSameSite);
+
+              return true;
+            }
+
+            /**
+             * Removes consent for a feature that has been remembered in a cookie
+             *
+             * @param {string} feature
+             * @return {boolean}
+             */
+            this.forgetConsentGivenForFeature = function (feature) {
+              if (!this.hasConsentForFeatureBeenGiven(feature)) {
+                return false;
+              }
+
+              var index = indexOfArray(configHasConsentForFeature, feature);
+              configHasConsentForFeature.splice(index, 1);
+
+              deleteCookie(getCookieName('consent_feature_'+feature), configCookiePath, configCookieDomain);
```

Comment:
Should the cookie name perhaps have a pk_ or matomo_ prefix to be consistent with other tracker cookie names?

Reply:
It should be prefixed already because calling `getCookieName`